### PR TITLE
Add the JSON schema generator

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

File: releng/gen.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	resource "github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	triggers "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"os"
+	"encoding/json"
+)
+
+
+func dump(v interface{}, apiVersion string, kind string, list bool) {
```
Comment (lstocchi) [inline releng/gen.go:16]: It looks to me that you use the bool `list` only to create the correct `realKind` (so if you pass `PipelineRun` as kind and list is true the `realKind` will be `PipelineRunList`), so why don't you directly pass the list kind, such as `PipelineRunList`?

**Comment by evidolob** [comment]
I'm wondering, is there any way how to share this code between vscode and IJ plugins?
Because for vscode-tekton we also need to have this schemes, and I don't think that copy/paste this PR in to vscode-tekton is a good idea.

And when I try to build schemes, with `go build` inside `releng` folder I got:
```
go: downloading github.com/pkg/errors v0.9.1
go: downloading google.golang.org/protobuf v1.22.0
go: downloading golang.org/x/sys v0.0.0-20200327173247-9dae0f8f5775
gen.go:6:2: cannot find module providing package github.com/redhat-developer/tekton-jsongenerator/jsonschema: module github.com/redhat-developer/tekton-jsongenerator/jsonschema: git ls-remote -q origin in /Users/yevhen/go/pkg/mod/cache/vcs/7b95259a8dccf701413e6130c6047d721c0b44486fb6807ef2bd9d94d9ad9df5: exit status 128:
        remote: Invalid username or password.
        fatal: Authentication failed for 'https://github.com/redhat-developer/tekton-jsongenerator/'
```
@jeffmaury Is there some other way to build it?

**Comment by jeffmaury** [comment]
`go build gen.go` went fine for me

**Comment by evidolob** [comment]
@jeffmaury I still not able to build, build fails when on resolving import:
https://github.com/redhat-developer/intellij-tekton/pull/132/files#diff-d8f1739e4f0ad7e11f5df49b79a8cec5R6
It seems `github.com/redhat-developer/tekton-jsongenerator/jsonschema` go package do not exists, or not pushed, as when I go to [tekton-jsongenerator](https://github.com/redhat-developer/tekton-jsongenerator) I got `404` error page.

**Comment by jeffmaury** [comment]
@evidolob are you using a go modules enabled version of go as it is working on my laptop

**Comment by evidolob** [comment]
It seems yes, build fails with:
```
go: finding module for package github.com/redhat-developer/tekton-jsongenerator/jsonschema
gen.go:6:2: cannot find module providing package github.com/redhat-developer/tekton-jsongenerator/jsonschema: module github.com/redhat-developer/tekton-jsongenerator/jsonschema: git ls-remote -q origin in /Users/yevhen/go/pkg/mod/cache/vcs/7b95259a8dccf701413e6130c6047d721c0b44486fb6807ef2bd9d94d9ad9df5: exit status 128:
        fatal: could not read Username for 'https://github.com': terminal prompts disabled
Confirm the import path was entered correctly.
If this is a private repository, see https://golang.org/doc/faq#git_https for additional information.
```
Rest of the modules resolved/download successfully

**Comment by evidolob** [comment]
@jeffmaury I have problem with JSON Scheme it self, it seems that `ArrayOrString` object is not generated correctly,
now it definition:
```json
"ArrayOrString": {
      "required": [
        "Type",
        "StringVal",
        "ArrayVal"
      ],
      "properties": {
        "Type": {
          "type": "string"
        },
        "StringVal": {
          "type": "string"
        },
        "ArrayVal": {
          "items": {
            "type": "string"
          },
          "type": "array"
        }
      },
      "additionalProperties": false,
      "type": "object"
    },
```

But in terms of JSON Scheme this means that it is `Object` not a string or array of strings, that lead to false positive errors/warnings when IJ/VSCode do validation:

Screenshot 2020-06-24 at 15 20 36

IJ shows warnings on 100% valid pipeline definition.(Same has VSCode when I put  this scheme to it)

**Pull request merged**